geode-management/src/main/java/org/apache/geode/management/client/ClusterManagementServiceBuilder.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package org.apache.geode.management.client;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.ServiceLoader;
+
+import javax.net.ssl.HostnameVerifier;
+import javax.net.ssl.SSLContext;
+
+import org.springframework.http.client.ClientHttpRequestFactory;
+
+import org.apache.geode.annotations.Experimental;
+import org.apache.geode.management.api.ClusterManagementService;
+
+@Experimental
+public class ClusterManagementServiceBuilder {
+
+  public static JavaBuilder buildWithUrl() {
+    return (JavaBuilder) getServiceBuilder(JAVA_CLIENT_CONTEXT);
+  }
+
+  public static GeodeBuilder buildWithCache() {
+    return (GeodeBuilder) getServiceBuilder(GEODE_CONTEXT);
+  }
+
+  public static HttpRequestFactoryBuilder buildWithRequestFactory() {
+    return (HttpRequestFactoryBuilder) getServiceBuilder(JAVA_CLIENT_CONTEXT);
+  }
+
+  public interface Builder {
+    ClusterManagementService build();
+
+    String getContext();
+  }
+
+  private interface GenericBuilder<T> extends Builder {
+    T setCredentials(String username, String password);
+  }
+
+  public interface GeodeBuilder extends GenericBuilder<GeodeBuilder> {
+    GeodeBuilder setCache(Object cache);
+  }
```

Comment:
it seems like this builder does not need to be present in the stand-along management jar, since it is only relevant for non-standalone usage.  Can we move it to geode-core or somewhere so that it exists only when suitable, and then can properly accept the correct parameter type?

Reply:
when `ClusterManagementServiceBuilder.buildWithCache()` method is called, it needs to return a `GeodeBuilder`, since `ClusterManagementServiceBuilder` exists in geode-management, so is the return type. We had to do this if we want to keep "one entry point" for building the various ClusterManagementService.

Reply:
I don't like the idea of exposing methods in a context where they won't work. Also having `setCache` take an `Object` as param is not good.